src/neighbors.cc

```diff
@@ -571,17 +527,22 @@ neighbors_init(void)
                 debugs(15, DBG_IMPORTANT, "WARNING: Peer looks like this host." <<
                        Debug::Extra << "Ignoring cache_peer " << *thisPeer);
 
-                neighborRemove(thisPeer);
+                peersToRemove.push_back(thisPeer.get());
+                break; // avoid warning about (and removing) the same CachePeer twice
             }
         }
+
+        while (peersToRemove.size()) {
+            const auto p = peersToRemove.back();
+            peersToRemove.pop_back();
+            DeleteConfigured(p);
+        }
```

Comment:
... with accompanying:
```suggestion
                Config.peers->erase(thisPeer);
                break; // avoid warning about (and removing) the same CachePeer twice
            }
        }
```